src/lib/support/ScopedBuffer.h

```diff
@@ -263,6 +263,23 @@ class ScopedMemoryBufferWithSize : public ScopedMemoryBuffer<T>
         return *this;
     }
 
+    // Allow creating ScopedMemoryBufferWithSize from CharSpan, ByteSpan, so we
+    // don't have to reinvent it in various places.
+    template <class U, typename = std::enable_if_t<sizeof(U) == sizeof(T) && std::is_convertible<U *, T *>::value>>
+    ScopedMemoryBufferWithSize & CopyFromSpan(const chip::Span<const U> & span)
+    {
+        ScopedMemoryBufferWithSize<T>::Alloc(span.size());
+        memcpy(ScopedMemoryBuffer<T>::Get(), span.data(), AllocatedSize());
```

Comment:
https://github.com/project-chip/connectedhomeip/pull/38749#discussion_r2100765589

Reply:
Checking span.size, freeing and returning if 0.

Reply:
@doru91 on conversation ask me to resolve.